src/core/lib/transport/error_utils.cc

```diff
@@ -52,7 +53,19 @@ void grpc_error_get_status(grpc_error* error, grpc_millis deadline,
   if (GPR_LIKELY(error == GRPC_ERROR_NONE)) {
     if (code != nullptr) *code = GRPC_STATUS_OK;
     if (slice != nullptr) {
-      grpc_error_get_str(error, GRPC_ERROR_STR_GRPC_MESSAGE, slice);
+      // Normally, we call grpc_error_get_str(
```

Comment:
If we really want to fast-path this case, since we've already checked that the error is `GRPC_ERROR_NONE`, how about just returning a static slice with value ""?  I'd be a lot happier with leaking that assumption outside of the error implementation than I am about leaking the error status map.

Reply:
Functionally it amounts to the same thing; we'd just need to safeguard in case GRPC_ERROR_NONE ever yields something other than "". It would allow us to re-privatize the special_error_status_map as well.

Perhaps: marshall a static string "" as you mention, then have an #ifndef NDEBUG wrapped function that asserts that it's equal to what the slowpath would have bought us?

Reply:
Fixed, using the newly added grpc_slice_from_static_string_internal.